File: LeetCode-30DayChallenge/MaximumSubArray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max_current = nums[0], max_global = nums[0];
        for (int i = 1; i < nums.length; i++) {
            max_current = Math.max(nums[i], nums[i] + max_current);
            max_global = Math.max(max_current, max_global);
        }
        return max_global;
    }

    public static void main(String ar[]) {
        Solution sol = new Solution();
        System.out.println(sol.maxSubArray(new int[] { -2, 1, -3, 4, -1, 2, 1, -5, 4 }));
    }
}

File: LeetCode-30DayChallenge/Go/MaximumSubarray.go
package main

import (
	"fmt"
)

func getMax(a int,b int) int{
    if(a > b){
        return a;
    }   
    return b;
}

func maxSubArray(nums []int) int {
    maxCurrent := nums[0]
    maxGlobal := nums[0]
    for i := 1;i<len(nums);i++{
        maxCurrent = getMax(nums[i],maxCurrent + nums[i])
        maxGlobal = getMax(maxGlobal,maxCurrent)
    }
    return maxGlobal
}

func main(){
	result := maxSubArray([]int{-2,1,-3,4,-1,2,1,-5,4})
	fmt.Println(result)
}